### DISCUSSION TITLE
How to create a DataPortalException object with HttpErrorInfo

### DISCUSSION THREAD

**Discussion by oleksandr-yevchenko** [MarimerLLC/csla#2367] [Questions]
Hi,
I am writing unit tests for handling exceptions in my solution using CSLA 5.4.1. And I need to replicate the behaviour of the app when it is handling DataPortalException.

So, I would like to create a DataPortalException using the constructor:
  ```
public DataPortalException(Csla.Server.Hosts.HttpChannel.HttpErrorInfo info)
        : base(info.Message)
      {
        this.ErrorInfo = info;
      }
```
    
   When I create it this way:
`    var ex = new DataPortalException(new HttpErrorInfo());`
    
I am getting the error hint on `new HttpErrorInfo()`:
    CSLA0011: CSLA business objects should not be created outside of a ObjectFactory instance.

Some research:
    1. `var errorInfo = DataPortal.Create<HttpErrorInfo>(new ValidationException("The object is not valid"));`
throws Csla.DataPortalException: 'DataPortal.Create failed (Invalid operation - create not allowed)'

 2. Disabling the warning works but I do not like writing the code like this:
```
#pragma warning disable CSLA0011
            HttpErrorInfo errorInfo = new HttpErrorInfo(new ValidationException("The object is not valid"));
#pragma warning restore CSLA0011
```

Thanks and cheers,
      Oleksandr

**Comment by rockfordlhotka** [comment]
You can safely use `pragma` in this case. You are encountering a case where the analyzer is catching the creation of a "business type", when in fact you are creating a type that isn't _intended_ to be created outside CSLA at all (other than for testing).
